src/asm/lexer.c

```diff
@@ -156,25 +156,39 @@ YY_BUFFER_STATE yy_create_buffer(FILE *f)
 	if (pBuffer == NULL)
 		fatalerror("%s: Out of memory!", __func__);
 
-	uint32_t size;
+	size_t size = 0, capacity;
+	char *buf = NULL;
 
 	fseek(f, 0, SEEK_END);
-	size = ftell(f);
-	fseek(f, 0, SEEK_SET);
+	capacity = ftell(f);
+	rewind(f);
 
-	/* Give extra room for 2 newlines and terminator */
-	uint32_t capacity = size + 3;
+	if (capacity == -1)
```

Comment:
`ftell` will not return EOF. If the end of file has been reached, the size of the file will be returned. `ftell` is specified to return `-1` in case of error, see `ftell(3)` or `ftell(3p)`

Reply:
Ah, apologies, it seems like I was looking at [duff information on cppreference.com](https://en.cppreference.com/mwiki/index.php?title=c/io/ftell&oldid=78945) (though, strangely, [the corresponding C++ page gets it right](https://en.cppreference.com/w/cpp/io/c/ftell)).

Either way, thanks for looking into resolving the `capacity == 0` issue! It probably doesn't need to be a separate condition to the above, but whatever 🤷‍♂️